iotile_ext_cloud/iotile/cloud/apps/cloud_uploader.py

```diff
@@ -158,6 +158,8 @@ def download(self, trigger=None, acknowledge=True):
         else:
             self.logger.info("Not acknowledging readings from cloud per user request")
 
+        # Configure Downloader to not break up the report
+        self._con.config_streaming(max_packet=0xFFFFFFF0)
```

Comment:
This app is in a coretools package which means in needs to not have any dependencies on non-coretools packages.  The `config_streaming` method is provided by `iotile-support-con-nrf52832`.  That means that adding this call will make the CloudUploader break in the following cases:
- user does not have `iotile-support-con-nrf52832` installed
- user is not talking to an nRF52832 based POD but still wants to upload data from it (such as a node or other virtual device).  

The goal with CoreTools apps such as CloudUploader is to provide global implementations of specific use cases (in this case uploading streamer reports to the cloud) that do not rely on functionality provided by the specific proxies of any given device but instead directly call the relevant RPCs directly and work with any device that implements those RPCs.

That's why there are no other reference to any con-nrf52832 specific functionality in this class; note that all of the other RPCs that are invokes are done using the direct `self._con.rpc` method which all devices provide, not assuming that there will be a proxy module installed that provides a convenience python wrapper, see the `basic=True` parameter here:
https://github.com/iotile/coretools/pull/509/files#diff-64ff6b7ed94c181141c2d873c3fc7854L43

The required rpcs that would nee to be added to CloudUploader are documented here:
https://github.com/iotile/coretools/issues/508#issuecomment-422848894

Also, not all devices support configuring the streamer report length so this method needs to gracefully handle when the RPC is not implemented (and raises a HardwareError).

Reply:
Thanks for that feedback. Actually... cloud_uploader.py already implements set_report_size and handles the errors as needed.... I just needed to call that method.